### PR TITLE
Add an overload of `AddSystemFonts`

### PR DESCRIPTION
**Pull request by LeaFrock**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/Fonts/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Add `public static FontCollection AddSystemFonts(this FontCollection collection, Predicate<FontMetrics> match)` for filtering unwanted system fonts.

See #239 for details.

### CODE REVIEW

File: src/SixLabors.Fonts/FontCollectionExtensions.cs [hunk 1]
```diff
@@ -17,12 +19,31 @@ public static FontCollection AddSystemFonts(this FontCollection collection)
...
             }
 
             return collection;
         }
+
+        /// <summary>
+        /// Adds the fonts from the <see cref="SystemFonts"/> collection to this <see cref="FontCollection"/>.
+        /// </summary>
+        /// <param name="collection">The font collection.</param>
+        /// <param name="match">The System.Predicate delegate that defines the conditions of <see cref="FontMetrics"/> to add into the font collection.</param>
```
Comment (JimBobSquarePants) [inline src/SixLabors.Fonts/FontCollectionExtensions.cs:34]:
```suggestion
        /// <param name="match">The <see cref="System.Predicate"/> delegate that defines the conditions of <see cref="FontMetrics"/> to add into the font collection.</param>
```

Comment (LeaFrock) [reply src/SixLabors.Fonts/FontCollectionExtensions.cs]: CI error fixed. Should be `<see cref="Predicate{T}"/>`.

File: src/SixLabors.Fonts/FontCollectionExtensions.cs [hunk 2]
```diff
@@ -17,12 +19,31 @@ public static FontCollection AddSystemFonts(this FontCollection collection)
...
             return collection;
         }
+
+        /// <summary>
+        /// Adds the fonts from the <see cref="SystemFonts"/> collection to this <see cref="FontCollection"/>.
+        /// </summary>
+        /// <param name="collection">The font collection.</param>
+        /// <param name="match">The <see cref="Predicate{T}"/> delegate that defines the conditions of <see cref="FontMetrics"/> to add into the font collection.</param>
+        /// <returns>The <see cref="FontCollection"/> containing the system fonts.</returns>
+        public static FontCollection AddSystemFonts(this FontCollection collection, Predicate<FontMetrics> match)
```
Comment (JimBobSquarePants) [inline src/SixLabors.Fonts/FontCollectionExtensions.cs:36]: We'll need to add a unit test for this overload. Maybe one that checks that filters by culture and checks the collection.

Comment (LeaFrock) [reply src/SixLabors.Fonts/FontCollectionExtensions.cs:36]: There has already been an unit test for `AddSystemFonts()`. So just filter all the system fonts and check if the font collection is empty,

`collection.AddSystemFonts(p => false);`

I think it's enough to make sure that the custom filter works in the method.

**Review by JimBobSquarePants — APPROVED** [review]
LGTM, thanks for the submission 👍

**Pull request merged**